types/json-patch-gen/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// Type definitions for json-patch-gen 1.0
+// Project: https://github.com/gregsexton/json-patch-gen
+// Definitions by: Konstantin Rohde <https://github.com/RohdeK>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare function diff(obj1: object | null, obj2: object | null): diff.JsonPatch[];
+
+declare namespace diff {
+  type PatchOperation = "replace" | "add" | "remove";
+
+  interface JsonPatch {
+      op: PatchOperation;
+      path: string;
+      value: any;
+  }
+}
+
+export = diff;
```

Comment:
Since this can be used as a global in the browser, can you also add `export as namespace diff`?

Reply:
Ah, I thought this was automatic when diff is also refering to a namespace. Will be added immediately.